jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/Entropy32Bytes.kt

```diff
@@ -0,0 +1,34 @@
+package com.radixdlt.sargon.extensions
+
+import com.radixdlt.sargon.BagOfBytes
+import com.radixdlt.sargon.Bip39Entropy
+import com.radixdlt.sargon.Entropy28Bytes
+import com.radixdlt.sargon.Entropy32Bytes
+import com.radixdlt.sargon.NotarySignature
+import com.radixdlt.sargon.PublicKey
+import com.radixdlt.sargon.SignedIntentHash
+import com.radixdlt.sargon.androidNotarizeHashWithPrivateKeyBytes
+import com.radixdlt.sargon.androidNotaryKeyGetPublicKeyFromPrivateKeyBytes
+import com.radixdlt.sargon.newEntropy32BytesFromBytes
+
+fun Entropy32Bytes.Companion.random() = init(randomBagOfBytes(LENGTH))
+
+@Throws(SargonException::class)
+fun Entropy32Bytes.Companion.init(bytes: BagOfBytes): Entropy32Bytes =
+    newEntropy32BytesFromBytes(bytes = bytes)
+
+val Entropy32Bytes.Companion.LENGTH: Int
+    get() = 32
+
+fun Entropy32Bytes.asGeneral() = Bip39Entropy.EntropyOf32Bytes(this)
+
+@Throws(SargonException::class)
+fun Entropy32Bytes.notarize(signedIntentHash: SignedIntentHash): NotarySignature =
```

Comment:
Personally I would not create `notarize` as a method on `Entropy32Bytes`, I find it a bit strange. I think I would create my own `data class`: `NotaryPrivateKey` which has `notarize` and the public key func returning `Ed25519PublicKey`.

Reply:
You are right was thinking about it too.